Foundation/GTMLogger.m

```diff
@@ -505,31 +522,12 @@ @implementation GTMLogLevelFilter
 - (id)init {
   self = [super init];
   if (self) {
-    // Keep a reference to standardUserDefaults, avoiding a crash if client code calls
-    // "NSUserDefaults resetStandardUserDefaults" which releases it from memory. We are still
-    // notified of changes through our instance. Note: resetStandardUserDefaults does not actually
-    // clear settings:
-    // https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSUserDefaults_Class/index.html#//apple_ref/occ/clm/NSUserDefaults/resetStandardUserDefaults
-    // and so should only be called in test code if necessary.
-    userDefaults_ = [[NSUserDefaults standardUserDefaults] retain];
-    [userDefaults_ addObserver:self
-                    forKeyPath:kVerboseLoggingKey
-                       options:NSKeyValueObservingOptionNew
-                       context:nil];
-
     verboseLoggingEnabled_ = IsVerboseLoggingEnabled(userDefaults_);
   }
 
   return self;
 }
 
-- (void)dealloc {
```

Comment:
Does it make sense to keep this and assert that userDefaults_ is nil?

Reply:
Sounds like a good idea. What kind of assertion should we use? `NSAssert`, `NSException`, or something else? I’m thinking an `NSInternalInconsistencyException`.

Reply:
`NSAssert` should be fine since it would likely be an issue in some impl calling these methods.  You should be able to wrap the whole dealloc in `defined(NS_BLOCK_ASSERTIONS)` so it won't even exist when the `NSAssert` is going to be compiled away.

Reply:
Added an assertion using `_GTMDevAssert`. Thanks!

I’ll leave out the `#if !defined(NS_BLOCK_ASSERTIONS)` part because it adds extra maintenance cost (needs to be kept in sync with the implementation of `dealloc` and the implementation of the `_GTMDevAssert` macro) and extra code (including a clarifying comment) without much benefit in this case.